### PR TITLE
Add support to app certificate

### PR DESCRIPTION
**Pull request by toshetah**

### CODE REVIEW

File: azure-kusto-data/azure/kusto/data/request.py [hunk 1]
```diff
@@ -100,6 +106,27 @@ def with_aad_application_key_authentication(cls, connection_string, aad_app_id,
...
+    @classmethod
+    def with_aad_application_certificate_authentication(
+        cls, connection_string, aad_app_id, certificate, thumbprint
+    ):
+        """Creates a KustoConnection string builder that will authenticate with AAD application and
+        password.
+        :param str connection_string: Kusto connection string should by of the format:
+        https://<clusterName>.kusto.windows.net
+        :param str aad_app_id: AAD application ID.
+        :param str certificate: A PEM encoded certificate private key.
```
Comment (vladikbr) [inline azure-kusto-data/azure/kusto/data/request.py]: Huh???

File: azure-kusto-data/azure/kusto/data/request.py [hunk 2]
```diff
@@ -100,6 +106,27 @@ def with_aad_application_key_authentication(cls, connection_string, aad_app_id,
         kcsb[kcsb.ValidKeywords.application_key] = app_key
         return kcsb
 
+    @classmethod
+    def with_aad_application_certificate_authentication(
+        cls, connection_string, aad_app_id, certificate, thumbprint
+    ):
+        """Creates a KustoConnection string builder that will authenticate with AAD application and
+        password.
```
Comment (vladikbr) [inline azure-kusto-data/azure/kusto/data/request.py]: This is incorrect

File: azure-kusto-data/azure/kusto/data/request.py [hunk 3]
```diff
@@ -140,6 +167,24 @@ def application_key(self):
         """The application key to use for authentication when federated authentication is used"""
         return self._internal_dict.get(self.ValidKeywords.application_key)
 
+    @property
+    def application_certificate(self):
+        """A PEM encoded certificate private key."""
```
Comment (vladikbr) [inline azure-kusto-data/azure/kusto/data/request.py]: Isn't it just "X509 certificate"

File: azure-kusto-data/azure/kusto/data/request.py [hunk 4]
```diff
@@ -140,6 +167,24 @@ def application_key(self):
...
+        return self._internal_dict.get(self.ValidKeywords.application_certificate)
+
+    @application_certificate.setter
+    def application_certificate(self, value):
+        self[self.ValidKeywords.application_certificate] = value
+
+    @property
+    def application_certificate_thumbprint(self):
+        """hex encoded thumbprint of the certificate."""
```
Comment (vladikbr) [inline azure-kusto-data/azure/kusto/data/request.py]: Thumbprint is always a hexadecimal string

Comment (toshetah) [reply azure-kusto-data/azure/kusto/data/request.py]: I took it from here:
https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/ef3a8814540684b6fbd709a249dc7c522f94584a/adal/authentication_context.py#L245

File: azure-kusto-data/azure/kusto/data/request.py (hunk 3 above)
Comment (toshetah) [reply azure-kusto-data/azure/kusto/data/request.py]: https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/ef3a8814540684b6fbd709a249dc7c522f94584a/adal/authentication_context.py#L244

File: azure-kusto-data/azure/kusto/data/request.py (hunk 1 above)
Comment (toshetah) [reply azure-kusto-data/azure/kusto/data/request.py]: https://github.com/AzureAD/azure-activedirectory-library-for-python/blob/dev/adal/authentication_context.py#L237-L247

File: azure-kusto-data/azure/kusto/data/request.py (hunk 2 above)
Comment (toshetah) [reply azure-kusto-data/azure/kusto/data/request.py]: Fixed.

**Pull request merged**